### PR TITLE
Allow managementState to be null, default to Removed

### PR DESCRIPTION
**Pull request by jboyd01**
* bump vendor to pick up latest prereqs which include better support for handling empty managementState
* Add manifest to create cluster ServiceCatalogAPIServer resource with managementStage=Removed
* Add manifest to create ServiceCatalogAPIServer ClusterOperator

### CODE REVIEW

**Review by jmrodri — APPROVED** [review]
The two manifest files look correct. I see the state is Removed by default which is what you described. The warning in the workload_controller makes sense as well. I did NOT look at the vendor changes in this PR.

File: manifests/06_service-catalog-apiserver.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: operator.openshift.io/v1
+kind: ServiceCatalogAPIServer
+metadata:
+  name: cluster
+spec:
+  logLevel: "Normal"
+  managementState: Removed
```
Comment (bparees) [inline manifests/06_service-catalog-apiserver.yaml]: i'm pretty sure that by doing this, the CVO will start stomping this field back to this value if it's edited (no one will be able to change it to anything else)

if you look at other operators, they create the CR resource programmatically, not via manifests/CVO.

File: pkg/operator/workloadcontroller/workload_controller.go [hunk 2]
```diff
@@ -115,7 +115,9 @@ func (c ServiceCatalogAPIServerOperator) sync() error {
 		}
 		return nil
 	default:
-		c.eventRecorder.Warningf("ManagementStateUnknown", "Unrecognized operator management state %q", operatorConfig.Spec.ManagementState)
+		if len(operatorConfig.Spec.ManagementState) > 0 {
+			c.eventRecorder.Warningf("ManagementStateUnknown", "Unrecognized operator management state %q", operatorConfig.Spec.ManagementState)
+		}
```
Comment (bparees) [inline pkg/operator/workloadcontroller/workload_controller.go]: why isn't an empty management state also considered to be an unrecognized management state and reported as such?  seems like it's still an error condition in some fashion.

File: manifests/06_service-catalog-apiserver.yaml (hunk 1 above)
Comment (jboyd01) [reply manifests/06_service-catalog-apiserver.yaml]: It has worked on all my tests, but now that you say this, I've always disabled the CVO to get my own operator deployed.  What you are saying makes a ton of sense, thanks Ben, I'll revisit.

File: pkg/operator/workloadcontroller/workload_controller.go (hunk 2 above)
Comment (jboyd01) [reply pkg/operator/workloadcontroller/workload_controller.go]: Perhaps it should be.  This was my understanding in conversations with @derekwaynecarr, but perhaps that was just step along the path to getting Service Catalog into the CVO payload with it disabled.  We are now setting it to an initial state of removed.  I'll revisit this along with the rework above.

**Comment by jboyd01** [comment]
@bparees thanks for spotting this issue with the creation of of CR in manifests vs having the clusteroperator create it.  I've reworked and addressed the issue with null managementState as well.

**Comment by bparees** [comment]
pr description is inaccurate re: null, otherwise lgtm.

**Pull request merged**